### PR TITLE
fix: 修复 pagination 组件属性透传问题, 简化 usePagination 逻辑

### LINKED ISSUE #1718: 🐛 分页组件无法调整 每页显示数
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       |     1.27.0     |
| @antv/s2-react |     1.25.0     |
| @antv/s2-vue   |          |

### Sheet Type

- [ ] PivotSheet
- [-] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet

### 🖋 Description
升级到最新版本后， 无法调整pageSize, 设置defaultPageSize 也不生效

### ⌨️ Code Snapshots
https://user-images.githubusercontent.com/46083022/187412696-1619daf1-5e94-4559-8070-479fc465384e.mov

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1718 , ref: #1698 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react/playground/index.less [hunk 1]
```diff
@@ -18,4 +18,8 @@
   .ant-space {
     flex-wrap: wrap;
   }
+
+  .ant-tabs-tabpane {
```
Comment (wjgogogo) [inline packages/s2-react/playground/index.less:22]: playground tab增加滚动，不然分页器会被遮住
image

**Comment by lijinke666** [comment]
加一个透传的测试吧, 避免 current, pageSize 这种被覆盖了

**Comment by wjgogogo** [comment]
@lijinke666 有这种情况的单测

**Review by lijinke666 — APPROVED** [review]

File: packages/s2-react/playground/index.less (hunk 1 above)
Comment (wjgogogo) [reply packages/s2-react/playground/index.less:22]: 是的

**Pull request merged**